### PR TITLE
MertonModelDescriptor + adjstuments in MertonModel Monte Carlo

### PR DESCRIPTION
**Pull request by AlessandroGnoatto**
### What is this PR for?
Add MertonModelDescriptor extend some factories. Introduce discount curves in the Merton Model class

### What type of PR is it?
Integrate the Merton Model in the structure allowing for separation of model and implementation.

### Todos
I noticed that Merton Model is parametrized in terms of doubles while Heston model in terms of RandomVariable... should we have more consistency here?

### What is the related issue?
Missing descriptor for the Merton Model. Missing discount curves in the Monte Carlo implementation of the Merton Model.

### Questions:

**Does the licenses files need update?**
NO

**Are there breaking changes for older versions?**
NO

**Does the change require additional documentation?**
NO

### CODE REVIEW

**Comment by AlessandroGnoatto** [comment]
Just a quick note: I did some refactoring work on all Fourier models. There were a lot of inconsistencies between different classes. What I saw:

- Missing Javadoc in model Classes.
- The field referenceDate was featured only in HestonModel and missing in the rest of the classes.
- BatesModel missed the DiscountCurves for forward rates and discounting.
- Constructor of the classes were a bit confusing, now we should have the same structure for all classes in the package.

File: src/main/java/net/finmath/fouriermethod/models/BlackScholesModel.java [hunk 1]
```diff
@@ -97,4 +126,63 @@ private double getLogDiscountFactorForDiscounting(double time) {
 		return discountCurveForDiscountRate == null ? -discountRate * time : Math.log(discountCurveForDiscountRate.getDiscountFactor(null, time));
 	}
 
+	/**
+	 * @return the referenceDate
+	 */
+	public LocalDate getReferenceDate() {
+		return referenceDate;
```
Comment (cfries) [inline src/main/java/net/finmath/fouriermethod/models/BlackScholesModel.java]: I would suggest the following for "backward compatibility": if referenceDate is null but the model has a discountCurve, then this model should return discountCurve.getReferenceDate().

File: src/main/java/net/finmath/fouriermethod/models/BatesModel.java [hunk 2]
```diff
@@ -94,23 +106,26 @@
 	 * @param delta Jump size variance.
 	 */
 	public BatesModel(
+			LocalDate referenceDate,
```
Comment (cfries) [inline src/main/java/net/finmath/fouriermethod/models/BatesModel.java]: Its maybe good to keep the old constructor for backward compatibility. (For example: I am not sure if the spreadsheets use this constructor?!). Its OK to put the old constructor to `@deprecated` .  Guideline is that version X.Y.(Z++) and X.Y(++).Z do not break old code. But X++ is allowed to break code and we can remove @deprecated code then.

Comment (AlessandroGnoatto) [reply src/main/java/net/finmath/fouriermethod/models/BatesModel.java]: Fine for me, thanks!

File: src/main/java/net/finmath/fouriermethod/models/BlackScholesModel.java (hunk 1 above)
Comment (AlessandroGnoatto) [reply src/main/java/net/finmath/fouriermethod/models/BlackScholesModel.java]: Fine for me, thanks!

**Pull request merged**